applications/GeoMechanicsApplication/custom_elements/line_2d_5.hpp

```diff
@@ -0,0 +1,1314 @@
+        {
+            return 4;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /** FacesNumber
+         * @return SizeType containes number of this geometry edges/faces.
+         */
+        SizeType FacesNumber() const override
+        {
+            return EdgesNumber();
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        ///@}
+        ///@name Shape Function
+        ///@{
+        /**
+         * @brief This method gives all non-zero shape functions values evaluated at the rCoordinates provided
+         * @note There is no control if the return vector is empty or not!
+         * @return Vector of values of shape functions \f$ F_{i} \f$ where i is the shape function index (for NURBS it is the inde of the local enumeration in the element).
+         * @see ShapeFunctionValue
+         * @see ShapeFunctionsLocalGradients
+         * @see ShapeFunctionLocalGradient
+         */
+        Vector& ShapeFunctionsValues(Vector& rResult, const CoordinatesArrayType& rCoordinates) const override
+        {
+            if (rResult.size() != 5) rResult.resize(5, false);
+            //
+            double xcor = rCoordinates[0];
+            double fx1 = xcor - 1.0;
+            double fx2 = xcor + 1.0;
+            double fx3 = fx1 * fx2;
+            double gx1 = 2.0 * xcor - 1.0;
+            double gx2 = 2.0 * xcor + 1.0;
+            double gx3 = gx1 * gx2;
+            double cof1 = 1.0 / 6.0;
+            double cof2 = 8.0 * cof1;
+            //
+            rResult[0] = xcor * fx1 * gx3 * cof1;
+            rResult[1] = xcor * fx2 * gx3 * cof1;
+            rResult[2] = -xcor * fx3 * gx1 * cof2;
+            rResult[3] = fx3 * gx3;
+            rResult[4] = -xcor * fx3 * gx2 * cof2;
+            //
+            return rResult;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * @brief This method gives value of given shape function evaluated in given point.
+         * @param rPoint Point of evaluation of the shape function. This point must be in local coordinate.
+         * @param ShapeFunctionIndex index of node which correspounding shape function evaluated in given integration point.
+         * @return Value of given shape function in given point.
+         * @see ShapeFunctionsValues
+         * @see ShapeFunctionsLocalGradients
+         * @see ShapeFunctionLocalGradient
+         */
+        double ShapeFunctionValue(IndexType ShapeFunctionIndex, const CoordinatesArrayType& rPoint) const override
```

Comment:
Shouldn't be const IndexType ShapeFunctionIndex?

Reply:
Done